modules/local/collect_stats.nf

```diff
@@ -0,0 +1,132 @@
+process COLLECT_STATS {
+    tag "$meta.id"
+    label 'process_low'
+
+    conda "conda-forge::r-tidyverse=2.0.0 conda-forge::r-dtplyr=1.3.1 conda-forge::r-data.table=1.14.8"
+    container "${ workflow.containerEngine == 'singularity' && !task.ext.singularity_pull_docker_container ?
+        'https://depot.galaxyproject.org/singularity/mulled-v2-b2ec1fea5791d428eebb8c8ea7409c350d31dada:a447f6b7a6afde38352b24c30ae9cd6e39df95c4-1' :
+        'biocontainers/mulled-v2-b2ec1fea5791d428eebb8c8ea7409c350d31dada:a447f6b7a6afde38352b24c30ae9cd6e39df95c4-1' }"
+
+    input:
+    tuple val(meta), val(samples), path(trimlogs), path(bblogs), path(idxstats), path(fcs)
+
+    output:
+    path "${meta.id}.overall_stats.tsv.gz", emit: overall_stats
+    path "versions.yml"                   , emit: versions
+
+    when:
+    task.ext.when == null || task.ext.when
+
+    script:
+    def args = task.ext.args ?: ''
+    def prefix = task.ext.prefix ?: "${meta.id}"
+
+    if ( trimlogs ) {
+        read_trimlogs = """%>%
+        mutate(
+            d = map(
+                sample,
+                function(s) {
+                    fread(
+                        cmd = sprintf("grep 'Reads written (passing filters)' %s*trimming_report.txt | sed 's/.*: *//' | sed 's/ .*//' | sed 's/,//g'",
+                        s)
+                    ) %>%
+                        as_tibble()
+                }
+            )
+        ) %>%
+        unnest(d) %>%
+        rename(n_trimmed = V1) %>%
+        mutate(n_trimmed = n_trimmed*2) %>%
+        """
+    } else {
+        read_trimlogs = "%>%"
+    }
+
+    """
+    #!/usr/bin/env Rscript
```

Comment:
These are getting increasingly long, I would make them into proper scripts in `bin/` (with a license at the top) and call the scripts here

Reply:
I see the advantages of what you propose @jfy133 , but there's a drawback, IMO, to having scripts in `bin/` as it's opaque to users when debugging an exception. If the whole script is in `.command.sh`, it's easy to see what went wrong (well, maybe), but if `.command.sh` just says `my_r_script.R` it's far more difficult. However, we're happy to follow conventions.